### ISSUE TITLE
JPA Criteria fails for values containing a dot

### ISSUE THREAD

**Issue by uladzimirFilipchanka** [Impetus/kundera#751]
We have a problem executing JPA Criteria. Following code fails with error "Arithmetic expression is not supported currently".

```
    CriteriaBuilder criteriaBuilder = getEM().getCriteriaBuilder();
    CriteriaQuery<Customer> query = criteriaBuilder.createQuery(Customer.class);
    Root<Customer> from = query.from(Customer.class);
    query.select(from.alias("cus"));
    query.where(criteriaBuilder.equal(from.get("firstName"), firstName));
    query.where(criteriaBuilder.equal(from.get("email"), email));
    query.where(criteriaBuilder.equal(from.get("lastName"), lastName));
    TypedQuery<Customer> q = getEM().createQuery(query);
```

It fails only when lastName variable contains a dot. And only lastname. Any other value for others field works fine.
For example, if we will change query and find against firstName, then lastName and them email, it will fail when an email contains a dot. 
It means that only last where clause doesn't allow a dot inside.

Reproduces only in 2.17. Previous version 2.16 works fine.

Meaningful part of stacktrace:

```
    com.impetus.kundera.KunderaException: Arithmetic expression is not supported currently
    at com.impetus.kundera.query.KunderaQueryUtils.getValue(KunderaQueryUtils.java:267) [kundera-core.jar:]
    at com.impetus.kundera.query.KunderaQueryUtils.onComparisonExpression(KunderaQueryUtils.java:557) [kundera-core.jar:]
    at com.impetus.kundera.query.KunderaQueryUtils.traverse(KunderaQueryUtils.java:406) [kundera-core.jar:]
    at com.impetus.kundera.query.KunderaQuery.initFilter(KunderaQuery.java:637) [kundera-core.jar:]
    at com.impetus.kundera.query.KunderaQuery.postParsingInit(KunderaQuery.java:541) [kundera-core.jar:]
    at com.impetus.kundera.query.QueryResolver.getQueryImplementation(QueryResolver.java:80) [kundera-core.jar:]
    at com.impetus.kundera.persistence.PersistenceDelegator.getQueryInstance(PersistenceDelegator.java:560) [kundera-core.jar:]
    at com.impetus.kundera.persistence.PersistenceDelegator.createQuery(PersistenceDelegator.java:521) [kundera-core.jar:]
    at com.impetus.kundera.persistence.EntityManagerImpl.createQuery(EntityManagerImpl.java:384) [kundera-core.jar:]
    at com.impetus.kundera.persistence.EntityManagerImpl.createQuery(EntityManagerImpl.java:722) [kundera-core.jar:]
    at com.impetus.kundera.persistence.EntityManagerImpl.createQuery(EntityManagerImpl.java:709) [kundera-core.jar:]
```

**Comment by karthikprasad13** [comment]
@uladzimirFilipchanka 

This is a known issue. We are marking it as a bug and will fix it in the next release. As a work around, you can append your string in `single quotes` for now.

```
query.where(criteriaBuilder.equal(from.get("lastName"), "'" + lastName + "'"));
```

-Karthik

**Comment by chhavigangwal** [comment]
Fixed with Kundera 3.0 release.

**Comment by chhavigangwal** [comment]
Fixed with Kundera 3.0 release.
